// Copyright 2025 MozoLM Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Message encapsulating configuation options for PPM model.

syntax = "proto3";

package mozolm;

option java_outer_classname = "PpmAsFstOptionsProto";

// Next available ID: 7
message PpmAsFstOptions {
  // Maximum order for the model.  Uses default if not set.
  int32 max_order = 1;

  // Alpha hyper-parameter for the PPM.  Uses default if not set.
  double alpha = 2;

  // Beta hyper-parameter for the PPM.  Uses default if not set.
  double beta = 3;

  // Whether the model is to be dynamic or static.
  bool static_model = 4;

  // Whether the model to be initialized is in Fst format, otherwise text.
  bool model_is_fst = 5;

  // Maximum number of states to cache. Uses default if not set.
  int64 max_cache_size = 6;
}
